lib/node_modules/@stdlib/blas/base/dtrsm/lib/base.js

```diff
@@ -0,0 +1,257 @@
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MAIN //
+
+/**
+* Solves matrix equation `op(A) * X = alpha * B` or `X * op(A) = alpha * B` where `alpha` is a scalar, `X` and `B` are m by n matrices, `A` is a unit, or non-unit, upper or lower triangular matrix and `op( A )` is one of `op( A ) = A` or `op( A ) = A ** T`. The matrix `X` is overwritten on `B`.
+*
+* @private
+* @param {string} order - storage layout of `A` and `B`
+* @param {string} side - specifies whether `op( A )` appears on the left or right of `X`
+* @param {string} uplo - specifies whether the upper or lower triangular part of the matrix `A` is supplied
+* @param {string} transa - specifies the form of `op( A )` to be used in matrix multiplication
+* @param {string} diag - specifies whether or not `A` is unit triangular
+* @param {NonNegativeInteger} m - number of rows in `B`
+* @param {NonNegativeInteger} n - number of columns in `B`
+* @param {number} alpha - scalar constant
+* @param {Float64Array} A - input matrix `A`
+* @param {NonNegativeInteger} LDA - stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`)
+* @param {NonNegativeInteger} offsetA - starting `A` index
+* @param {Float64Array} B - input matrix `B`
+* @param {NonNegativeInteger} LDB - stride of the first dimension of `B` (a.k.a., leading dimension of the matrix `B`)
+* @param {NonNegativeInteger} offsetB - starting `B` index
+* @returns {Float64Array} `B`
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 3.0, 0.0, 4.0 ] );
+* var B = new Float64Array( [ 5.0, 7.0, 0.0, 8.0 ] );
+*
+* dtrsm( 'row-major', 'left', 'upper', 'none', 'non-unit', 2, 2, 6.0, A, 2, 0, B, 2, 0 );
+* // B => <Float64Array>[ 30.0, 6.0, 0.0, 12.0 ]
+*/
+function dtrsm( order, side, uplo, transa, diag, m, n, alpha, A, LDA, offsetA, B, LDB, offsetB ) { // eslint-disable-line max-len, max-params
+	var nounit;
+	var lside;
+	var upper;
+	var tmp;
+	var i;
+	var j;
+	var k;
+
+	lside = side === 'left';
+	nounit = diag === 'non-unit';
+	upper = uplo === 'upper';
+
+	if ( m === 0 || n === 0 ) {
+		return B;
+	}
+	if ( order === 'row-major' ) {
+		lside = !lside;
+		upper = !upper;
+		order = 'column-major';
+	}
+	if ( alpha === 0.0 ) {
+		for ( j = 0; j < n; j++ ) {
+			for ( i = 0; i < m; i++ ) {
+				B[ offsetB + ( j * LDB ) + i ] = 0.0;
+			}
+		}
+		return B;
+	}
+	if ( lside ) {
+		if ( transa === 'none' ) {
+			// B := alpha * inv( A ) * B
+			if ( upper ) {
+				for ( j = 0; j < n; j++ ) {
+					if ( alpha !== 1.0 ) {
+						for ( i = 0; i < m; i++ ) { // eslint-disable-line max-depth
```

Comment:
Rather than disable the lint rule at each depth, just go ahead and disable for the file above `'use strict'`.